### ISSUE TITLE
Add clever element navigator

### ISSUE THREAD

**Issue by MikaelMayer** [MikaelMayer/Editor#9]
## Current state of element selection.

Elements can be selected in two ways. Either by clicking directly, or moving the selection around. Currently, moving the selection around is done using this:

![image](https://user-images.githubusercontent.com/3601079/58587087-20a9ab00-8222-11e9-97b4-b7c7f7083d64.png)

The icon on the left selects the parent, the next icon the previous sibling, the next icon the next sibling, and the last icon the first child.
The tag name of the selected element is below, and can be modified. The preview of the 

## Problems with the current selection:

* Based on user feedback, previous (resp. next) sibling selection should be a arrow (resp right) arrow, and parent selection should be an up arrow.
* Because there is not always a child or a sibling, these icons appear and disappear, making it very difficult to navigate smoothly. For example, the arrow for the previous sibling turns into an arrow for the next sibling when the first child is reached.

## Better solution

A 2-state interface to display a node with its children. depending on the context, the selected node can be either display as a child (if it does not have any children) or as the parent (if it has children)
Something like this:
```
------------------------------------
|     Parent Node tag/id/class     |
|----------------------------------|
| child(n-1) [T] child(n) [T] child(n+1)   |
|__________________________________|
```
If there is some non-space text between these children elements, there should be some selectable by a T icon indicating this between children that, if clicked, selects the trimmed text.

### Interface number 1 (by default for elements having child nodes)
* The surrounding box is in selected color (e.g. orange gray), all children are grey
* If click on the outside box, the interface changes to interface number 2
* If click on a child, this child becomes selected and the interface stays in number 1 mode unless the child does not have child nodes.

### Interface number 2 (by default for elements without child nodes, like text nodes or title)
* The surrounding box is in unselected colod, the child(n) is in selected color, all other children are gray
* If we click on the selected child(n) and it has children, the selection stays the same but the interfaces switches to number 1
* If we click on another child, even if it has  children, it should stay in number 2.

Animations are not needed for now.

**Comment by MikaelMayer** [comment]
@eou will fix this issue when his pull request #15 will be merged

**Comment by MikaelMayer** [comment]
Implemented in #15
